core/src/main/kotlin/org/jetbrains/research/testspark/core/generation/llm/network/RequestManager.kt

```diff
@@ -75,7 +78,7 @@ abstract class RequestManager(var token: String) {
             return LLMResponse(ResponseErrorCode.EMPTY_LLM_RESPONSE, null)
         }
 
-        val testSuiteGeneratedByLLM = testsAssembler.assembleTestSuite(packageName)
+        val testSuiteGeneratedByLLM = testsAssembler.assembleTestSuite(packageName, language)
```

Comment:
If we manage to re-define the structure of code in a way that `TestsAssembler` does not know about the `Language`, we will not need to propagate this `langauge` variable in the chain `LLMWithFeedbackCycle -> RequestManager -> TestsAssembler`.

Reply:
It would be cool to figure out all language specifics on the very top code level and make the feedback cycle work merely with the abstractions, leaving it unaware of language details.

Reply:
And even better to make it the testing framework agnostic, as well.
I will try to refactor the code in this sense before the end of my internship, thank you a lot for the detailed review!